cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -90,6 +91,66 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Translates a Python [`With`](https://docs.python.org/3/library/ast.html#ast.With) into a
+     * [Block].
+     *
+     * We return a Block to handle the with statement, following
+     * [PEP 343](https://peps.python.org/pep-0343/#specification-the-with-statement). The context
+     * managers `__enter__()` runs before the try block and `__exit__()` is always called later in
+     * the `finally` block to clean up, whether an exception occurs or not.
+     */
+    private fun handleWithStatement(node: Python.AST.With): Block {
+        val mainBlock = newBlock().implicit()
+        val exitStatements = mutableListOf<Statement>()
+        val tryBlock = newBlock().implicit()
```

Comment:
code and loc

Reply:
What would you expect here? The whole while or only the body?

Reply:
Given your "translation" in the doc string: only the `body` as this would be the closest match.